### DISCUSSION TITLE
speedtest: failed to read

### DISCUSSION THREAD

**Discussion by mircotronchin** [alexjustesen/speedtest-tracker#1072] [Q&A]
hello,
speedtest-tracker 0.14.4.
Speedtest worked fine for month, but has been failing for 2 days.
if I run speedtest from the container the test starts, but after 2 or 3 second it fails. this is the result:

root@30beec7e5a49:/var/www/html# speedtest -s 5015
 Speedtest by Ookla
 Server: TIM SpA - Padova (id: 5015)
 ISP: Telecom Italia
Idle Latency:     9.97 ms   (jitter: 0.17ms, low: 9.87ms, high: 10.11ms)
Download: FAILED                                            cy: 35.02 ms
[error] Cannot read:

I've deleted and redeployed speedtest-tracker container, but had the same problem.
Changing speedtest server doesn't solve the problem.
How to solve?

thank you.

edit: from outside of the container speedtest work fine:

ubuntu@ubuntu:~$ speedtest -s 5015
Speedtest by Ookla
Server: TIM SpA - Padova (id: 5015)
ISP: Telecom Italia
Idle Latency:     9.53 ms   (jitter: 0.90ms, low: 9.35ms, high: 11.01ms)
Download:   947.11 Mbps (data used: 1.1 GB)
32.20 ms   (jitter: 16.63ms, low: 8.11ms, high: 545.03ms)
Upload:   104.15 Mbps (data used: 47.0 MB)
113.94 ms   (jitter: 34.57ms, low: 10.53ms, high: 186.70ms)
Packet Loss:     0.0%
  
ubuntu@ubuntu:~$ speedtest -s 58567
Speedtest by Ookla
Server: TIM SpA - Verona (id: 58567)
ISP: Telecom Italia
Idle Latency:     6.51 ms   (jitter: 0.20ms, low: 6.39ms, high: 6.63ms)
Download:   940.55 Mbps (data used: 1.0 GB)
19.81 ms   (jitter: 10.61ms, low: 5.61ms, high: 543.97ms)
Upload:   104.14 Mbps (data used: 64.5 MB)
113.56 ms   (jitter: 32.74ms, low: 5.66ms, high: 180.97ms)
Packet Loss:     0.0%
  

edit 2: 
I've installed one new speedtest container, with new config volume. same error: inside the container speedtest fail, outside work fine.

edit 3: 
I've installed speedtest-tracker in a new container, with new Mariadb container. After some failed test, now it's working.. for now... 
I don't know the reason of this problem..

**Comment by mircotronchin** [comment]
after 2 days the same "download failed" error is returned.
I honestly I don't know how to resolve.
can anyone help me?

**Reply by alexjustesen** [reply]
Sometimes the server you're connecting to doesn't return good data. Try a different server and see if the issue continues.

**Reply by mircotronchin** [reply]
I don't think this is the reason: speedtest (from command line) fail inside the container but work fine from outside (same linux vm) or from other pc/mac with the same internet connection, with the nearest server or other servers.
BTW, it work now.. but I haven't changed nothing.

**Reply by alexjustesen** [reply]
🤷‍♂️ no clue then hahaha
